File: 0892.三维形体的表面积/0892-三维形体的表面积.py
class Solution:
    def surfaceArea(self, grid: List[List[int]]) -> int:
        all_cubic = sum(map(sum,grid))
        e = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])-1):
                e += min(grid[i][j],grid[i][j+1])
        for i in range(len(grid[0])):
            for j in range(len(grid)-1):
                e += min(grid[j][i],grid[j+1][i])
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                e += max(0,grid[i][j]-1)
        return all_cubic*6 - e*2
                

File: 0892.三维形体的表面积/0892-三维形体的表面积.rs
impl Solution {
    pub fn surface_area(grid: Vec<Vec<i32>>) -> i32 {
        let mut all_cubic: i32 = 0;
        let r = grid.len();
        let c = grid[0].len();
        for i in 0..r {
            for j in 0..c {
                all_cubic += grid[i][j];
            }
        }
        let mut e: i32 = 0;
        for i in 0..r {
            for j in 0..c {
                e += (grid[i][j] -1).max(0);
            }
        }
        for i in 0..r {
            for j in 0..c-1 {
                e += grid[i][j].min(grid[i][j+1]);
            }
        }
        for i in 0..c {
            for j in 0..r-1 {
                e += grid[j][i].min(grid[j+1][i]);
            }
        }
        
        return all_cubic*6 - e*2;
    }
}